### ISSUE TITLE
Convert option not available

### ISSUE THREAD

**Issue by IronSean** [MicrosoftDocs/visualstudio-docs#4917]
My WCF Services project doesn't list the Convert option when right clicking. How can I figure out why the option isn't available, or how can I manually achieve the same thing without the automated conversion?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c951b7f-f6df-042f-fbfe-6df71c5dbcdf
* Version Independent ID: fbf62d2c-c9b7-872e-d430-61df560e1509
* Content: [Migrate & publish web application to Azure Cloud Service - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/azure/vs-azure-tools-migrate-publish-web-app-to-cloud-service?view=vs-2019#feedback)
* Content Source: [docs/azure/vs-azure-tools-migrate-publish-web-app-to-cloud-service.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/azure/vs-azure-tools-migrate-publish-web-app-to-cloud-service.md)
* Product: **visual-studio-windows**
* Technology: **vs-azure**
* GitHub Login: @ghogen
* Microsoft Alias: **ghogen**

**Comment by WilliamAntonRohm** [comment]
@IronSean -- Sean, thank you for your question.  You may find an answer here:
- [https://social.msdn.microsoft.com/Forums/vstudio](https://social.msdn.microsoft.com/Forums/vstudio) 
- [https://www.stackoverflow.com](https://www.stackoverflow.com) 
- [https://www.reddit.com/r/VisualStudio/](https://www.reddit.com/r/VisualStudio) 

@ghogen -- Gordon, please look into this issue.

**Comment by IronSean** [comment]
In my case I _do_ already have another web project in the solution and I think I read somewhere that it only shows up if there are no web role projects pre-existing. 

The current solution looks like:
- Domain.csproj
- WFCServices.csproj
- MVCApp.csproj

I'm currently looking into just adding the services to the web app project so they can deploy all at once

**Comment by ghogen** [comment]
@sayedihashimi You might have an answer for this.

**Comment by sayedihashimi** [comment]
I'm not sure, perhaps @vijayrkn has some insight here?

**Comment by vijayrkn** [comment]
Checking with the right team.

**Comment by dbreshears** [comment]
This context menu was removed based on a a couple different issues that arose awhile back.   Duplicate Publish context menus after conversion and performance issues with unrelated projects.  The needed work to address these issues was deemed to costly based on the value the convert menu actually provided.  I apologize that the documentation wasn't addressed and that will need to be fixed.

To manually do the conversion.    With existing WCF service in solution

1. Right click and Add a new project to the solution and choose Cloud Service (classic).  Don't add any roles so the only project that will be created is the Cloud Service project.
2. Right click "Roles" of the Cloud Service project and choose "Add->Add Webrole project in solution" and select the WCF service project

Sorry for the confusion on this.

**Comment by IronSean** [comment]
Hi @dbreshears, thank you for the additional information. To follow up on this I ended up achieving what I wanted by first bringing the .svc files to my ASP.NET project and hosting them directly there, but then actually just converted them to normal API endpoints. This allowed me to achieve my end goal of getting both the ASP.NET application and the service methods deployed to Azure, and in one instance rather than two.

But I agree the the documentation should probably be updated to reflect that this UI was removed, and possibly reference the alternative steps you mentioned in your comment. Feel free to close this when it makes sense for you.
